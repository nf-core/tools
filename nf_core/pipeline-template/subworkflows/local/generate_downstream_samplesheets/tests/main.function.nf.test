
nextflow_function {

    name "Test Functions"
    script "../main.nf"
    tag 'subworkflows'
    tag 'generate_downstream_samplesheets'
    tag 'subworkflows/generate_downstream_samplesheets'

    test("Test Function channelToSamplesheet - csv") {

        function "channelToSamplesheet"

        when {
            function {
                """
                // define inputs of the function here. Example:
                input[0] = Channel.of(
                    [
                        sample: 'test-pe',
                        fastq_1: 'https://raw.githubusercontent.com/nf-core/test-datasets/modules/data/genomics/sarscov2/illumina/fastq/test_1.fastq.gz',
                        fastq_2: 'https://raw.githubusercontent.com/nf-core/test-datasets/modules/data/genomics/sarscov2/illumina/fastq/test_1.fastq.gz',
                        strandedness: 'auto'
                    ],
                    [
                        sample: 'test-se',
                        fastq_1: 'https://raw.githubusercontent.com/nf-core/test-datasets/modules/data/genomics/sarscov2/illumina/fastq/test_1.fastq.gz',
                        fastq_2: '',
                        strandedness: 'auto'
                    ])
                input[1] = "$outputDir/test.csv"
                input[2] = "csv"
                """
            }
        }

        then {
            assertAll(
                { assert function.success },
                { assert snapshot(function.result).match() }
            )
        }
    }

    test("Test Function channelToSamplesheet - tsv") {

        function "channelToSamplesheet"

        when {
            function {
                """
                // define inputs of the function here. Example:
                input[0] = Channel.of(
                    [
                        sample: 'test-pe',
                        fastq_1: 'https://raw.githubusercontent.com/nf-core/test-datasets/modules/data/genomics/sarscov2/illumina/fastq/test_1.fastq.gz',
                        fastq_2: 'https://raw.githubusercontent.com/nf-core/test-datasets/modules/data/genomics/sarscov2/illumina/fastq/test_1.fastq.gz',
                        strandedness: 'auto'
                    ],
                    [
                        sample: 'test-se',
                        fastq_1: 'https://raw.githubusercontent.com/nf-core/test-datasets/modules/data/genomics/sarscov2/illumina/fastq/test_1.fastq.gz',
                        fastq_2: '',
                        strandedness: 'auto'
                    ])
                input[1] = "$outputDir/test.tsv"
                input[2] = "tsv"
                """
            }
        }

        then {
            assertAll(
                { assert function.success },
                { assert snapshot(function.result).match() }
            )
        }
    }
}
