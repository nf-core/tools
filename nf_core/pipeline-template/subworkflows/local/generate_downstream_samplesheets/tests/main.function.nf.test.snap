{
    "Test Function channelToSamplesheet - tsv": {
        "content": [
            {
                
            }
        ],
        "meta": {
            "nf-test": "0.9.1",
            "nextflow": "24.10.0"
        },
        "timestamp": "2024-10-30T10:06:13.403158303"
    },
    "Test Function channelToSamplesheet - csv": {
        "content": [
            {
                
            }
        ],
        "meta": {
            "nf-test": "0.9.1",
            "nextflow": "24.10.0"
        },
        "timestamp": "2024-10-30T10:06:02.487840724"
    }
}